plugin/path_creds.go

```diff
@@ -198,11 +221,29 @@ func (b *backend) generateAndReturnCreds(ctx context.Context, engineConf *config
 	}
 	b.credCache.SetDefault(roleName, cred)
 
+	// Delete the WAL entry
+	if err := framework.DeleteWAL(ctx, storage, walID); err != nil {
+		// The rotation was successful, so don't return the error.
+		// The WAL will eventually be discarded by the rollback handler.
+		b.Logger().Warn("failed to delete password rotation WAL", "error", err.Error())
+	}
+
 	return &logical.Response{
 		Data: cred,
 	}, nil
 }
 
+// getUsername extracts the username from a service account name by
+// splitting on @. For example, if vault@hashicorp.com is the service
+// account, vault is the username.
+func getUsername(serviceAccount string) (string, error) {
+	fields := strings.Split(serviceAccount, "@")
+	if len(fields) > 0 {
+		return fields[0], nil
```

Comment:
Can the username be empty, and if not should we error? e.g. `@foo` returns `[]string{"", "foo"}`

Reply:
It errors right after this line, actually:
```return "", fmt.Errorf("unable to infer username from service account name: %s", serviceAccount)
```